### ISSUE TITLE
implement feature for newsletter-categories

### ISSUE THREAD

**Issue by sonority** [Ecodev/newsletter#44]
it would be nice if a frontend-user could register himself for specific newsletter-categories (ie. with help of "femanager" or similar frontend-user-registration, or simple included as part of the newsletter-system).

...are there any thoughts/ideas in this direction?

**Comment by marvin-martian** [comment]
I think it is fine as is. You can implement categories using tt_address or other address extensions. You can then create separate recipient lists via custom SQL to sort by category. I don't think the newsletter should deal with the frontend forms or the collection and management of addresses there is enough extensions out there that can do this.  By being able to write your own SQL to build recipients lists opens up the newsletter tool to all sorts of possibilities. 

If you really want to make the tool to be very flexible, allow the ability to configure recipients using separate shell scripts for selecting recipients, registering bounced mail, etc.. That way it can truly be implemented using any way possible. i.e. external DB's and not just MySQL, SOAP services etc.. That way a implementor could bind it with external applications such as say Magento or other CRM apps in separate environments.

Maybe you can set up a wiki on github where users can post their configuration examples with other extensions/applications. You are right there are too many permutations of how you could use this extension with other third-party extensions. A wiki is a good solution for this type of thing as what you can do with the extension is in continual flux. In the manual you could just point the users to the wiki.

**Comment by PowerKiKi** [comment]
I seemed to have missed the "category" aspect in my previous answer. I agree with @marvin-martian that categories should be implemented via RecipientList.

To avoid false expectation I close this issue, because I don't think anything will be done toward that. However I just opened the [wiki for this project](https://github.com/Ecodev/newsletter/wiki), if any of you have interesting info to share, feel free to create a page.
